@xen-orchestra/backups/_runners/VmsXapi.mjs

```diff
@@ -110,17 +120,31 @@ export const VmsXapi = class VmsXapiBackupRunner extends Abstract {
                     throttleStream,
                     vm,
                   }
-                  let vmBackup
-                  if (job.mode === 'delta') {
-                    vmBackup = new IncrementalXapi(opts)
-                  } else {
-                    if (job.mode === 'full') {
-                      vmBackup = new FullXapi(opts)
-                    } else {
-                      throw new Error(`Job mode ${job.mode} not implemented`)
+
+                  return pRetry(
+                    () => {
+                      let vmBackup
+                      switch (job.mode) {
+                        case 'delta':
+                          vmBackup = new IncrementalXapi(opts)
+                          break
+                        case 'full':
+                          vmBackup = new FullXapi(opts)
+                          break
+                        default:
+                          throw notImplementedJobModeError
+                      }
+
+                      return vmBackup.run()
+                    },
+                    {
+                      delays: fibonacci()
+                        .take(allSettings[''].nRetriesVmBackupFailures ?? 0)
+                        .toMs(),
+                      onRetry: () => debug('retry backup job', { jobId: job.id }),
```

Comment:
since a retry is quite important, I would show in the the UI, with a Task.warning, that way the users will be able to see a failing process